kamu-core/src/domain/repos/object_repository.rs

```diff
@@ -70,8 +70,8 @@ pub trait ObjectRepository: Send + Sync {
 
 #[derive(Clone, PartialEq, Eq, Debug)]
 pub struct InsertResult {
+    /// Hash of the inserted object
     pub hash: Multihash,
-    pub already_existed: bool,
```

Comment:
S3 does not support an atomic "insert if does not exist" operation, and I thought that checking with HEAD on every object write just to fill this flag would be too expensive (adds a whole extra round-trip).

So I decided to get rid of the flag altogether. In a situation where likelihood of large object already existing in the target repo - a higher level logic can do the pre-insert checks.